x-pack/plugins/enterprise_search/server/routes/app_search/analytics.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../plugin';
+
+const querySchema = {
+  start: schema.maybe(schema.string({ minLength: 10, maxLength: 10 })), // Date string, format 'YYYY-MM-DD'
+  end: schema.maybe(schema.string({ minLength: 10, maxLength: 10 })), // Date string, format 'YYYY-MM-DD'
+  tag: schema.maybe(schema.string()),
+};
+const queriesSchema = {
+  ...querySchema,
+  size: schema.maybe(schema.number({ max: 20 })),
+};
+
+export function registerAnalyticsRoutes({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/engines/{engineName}/analytics/queries',
+      validate: {
+        params: schema.object({
+          engineName: schema.string(),
+        }),
+        query: schema.object(queriesSchema),
+      },
+    },
+    async (context, request, response) => {
+      const { engineName } = request.params;
+
+      return enterpriseSearchRequestHandler.createRequest({
+        path: `/as/engines/${engineName}/analytics/queries`,
+      })(context, request, response);
+    }
+  );
+
+  router.get(
+    {
+      path: '/api/app_search/engines/{engineName}/analytics/queries/{query}',
```

Comment:
fyi I changed this slightly from the ent-search URL to be more RESTful (`analytics/query/xyz`->`analytics/queries/xyz`). It also follows the URL pattern we use for our documents API  (e.g.`engine/documents/id`, not `/engine/document/id`).